tests/lib/db_handler.py

```diff
@@ -0,0 +1,231 @@
+"""
+    Helper class for automatic testing, provides dump and restore via command
+    line utilities.
+
+    Expect mongodump, mongoimport and mongorestore present at PATH
+"""
+import os
+import pymongo
+import subprocess
+
+
+class DBHandler:
+
+    def __init__(self, uri=None, host=None, port=None,
+                 user=None, password=None):
+        """'uri' or rest of separate credentials"""
+        if uri:
+            self.uri = uri
+        if host:
+            if all([user, password]):
+                host = "{}:{}@{}".format(user, password, host)
+            self.uri = 'mongodb://{}:{}'.format(host, port or 27017)
+
+        assert self.uri, "Must have uri to MongoDB"
+        self.client = pymongo.MongoClient(uri)
```

Comment:
Missed a spot 😄 
If `uri` is None and `host` is used, should pass `self.uri` into MongoClient instead of `uri`.

Also I would suggest to change the if-statement clause into :
```python
if uri:
    ...
elif host:
    ...

```

Reply:
Changed